### PR TITLE
travis: validate CLA signers in a travis CI job

### PR DESCRIPTION
**Pull request by blackboxsw**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ install:
     - pip install tox
 
 script:
-    - tox
+  - tox
```
Comment (OddBloke) [inline .travis.yml]: This change is inconsistent with the rest of the file.
```suggestion
    - tox
```

File: tools/check-cla [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+fi
+cat <<EOF
+______________________________________________________________
+
+Hello $author,
+
+Thank you for your continued contributions to cloud-init and
+for signing the Contributor License Agreement for this project.
+
+Once CI passes, team will assign a reviewer to your branch
```
Comment (OddBloke) [inline tools/check-cla]:
```suggestion
Once CI passes, the cloud-init team will assign a reviewer to your branch
```

File: tools/check-cla [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+to help get your branch merged.
```
Comment (OddBloke) [inline tools/check-cla]:
```suggestion
to work on getting it merged.
```

File: tools/check-cla [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+author=${1/\/cloud-init/}
+if [ `grep -c ${author} tools/.lp-to-git-user` = 0 ]; then
+    cat <<EOF
+______________________________________________________________
+
+Hello $author,
+
+Thank you for your contribution to cloud-init.
+
+In order to accept this pull request. Please confirm
```
Comment (OddBloke) [inline tools/check-cla]: This first sentence looks incomplete. Perhaps:

```suggestion
In order to accept this pull request, you need to have signed the Contributor License Agreement (CLA). Please confirm
```

Comment (OddBloke) [reply tools/check-cla]: Also maybe s/confirm/ensure/ (as confirmation implies that they have signed it in the past, which won't be true for everyone).

File: tools/check-cla [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+that the Contributor License Agreement (CLA) has been signed
```
Comment (OddBloke) [inline tools/check-cla]: The CLA has certainly been signed, we specifically need it signed by this person:

```suggestion
that you have signed the CLA
```

(also dropped spelling out CLA as my suggestion for the previous line now includes it)

File: tools/check-cla [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    cat <<EOF
+______________________________________________________________
+
+Hello $author,
+
+Thank you for your contribution to cloud-init.
+
+In order to accept this pull request, you need to have signed
+the Contributor License Agreement (CLA). Please ensure that
+your have signed the CLA by following our hacking guide at:
```
Comment (OddBloke) [inline tools/check-cla]:
```suggestion
you have signed the CLA by following our hacking guide at:
```

File: tools/check-cla [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+author=${1/\/cloud-init/}
+if [ `grep -c ${author} tools/.lp-to-git-user` = 0 ]; then
+    cat <<EOF
+______________________________________________________________
+
+Hello $author,
+
+Thank you for your contribution to cloud-init.
+
+In order to accept this pull request, you need to have signed
```
Comment (OddBloke) [inline tools/check-cla]:
```suggestion
In order for us to merge this pull request, you need to have signed
```

"merge" feels more active that "accept".

File: tools/check-cla [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+the Contributor License Agreement (CLA). Please ensure that
+your have signed the CLA by following our hacking guide at:
+https://cloudinit.readthedocs.io/en/latest/topics/hacking.html
+______________________________________________________________
```
Comment (OddBloke) [inline tools/check-cla]: I wonder if we should leave a breadcrumb here for people who have signed the CLA but not transferred it to GH. They may think "I've signed the CLA, this message is just wrong" without clicking through to the HACKING doc at all. Maybe:

> (If you have previously signed the CLA, you will need to transfer your signature to GitHub. The hacking guide includes instructions on how to do this.)

**Comment by blackboxsw** [comment]
Closing as @OddBloke and I talked about his approach using github actions instead. Will re-open if need be

**Comment by OddBloke** [comment]
I think this is still valuable in the interim, the labels applied thus far have been manual.

**Comment by blackboxsw** [comment]
Converted this branch to use github actions https://github.com/canonical/cloud-init/pull/155

**Pull request closed without merging**